src/overlays/actors/ovl_En_Kusa/z_en_kusa.c

```diff
@@ -55,72 +111,656 @@ static InitChainEntry D_80936754[] = {
+
+        if (bug == NULL) {
+            break;
+        }
+    }
+}
+
+s32 EnKusa_GetWaterBox(EnKusa* this, GlobalContext* globalCtx) {
+    s32 pad;
+    WaterBox* waterBox;
+    f32 ySurface;
+    s32 bgId;
+
+    if (WaterBox_GetSurfaceImpl(globalCtx, &globalCtx->colCtx, this->actor.world.pos.x, this->actor.world.pos.z,
+                                &ySurface, &waterBox, &bgId) &&
+        (this->actor.world.pos.y < ySurface)) {
+        return true;
+    }
+    return false;
+}
+
+void EnKusa_InitCollider(Actor* thisx, GlobalContext* globalCtx) {
+    EnKusa* this = THIS;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    Collider_UpdateCylinder(thisx, &this->collider);
+}
+
+void EnKusa_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnKusa* this = THIS;
+    s32 pad;
+    s32 kusaType = GET_KUSA_TYPE(&this->actor);
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+
+    if (globalCtx->csCtx.state != 0) {
+        this->actor.uncullZoneForward += 1000.0f;
+    }
+    EnKusa_InitCollider(&this->actor, globalCtx);
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
+
+    if (kusaType == ENKUSA_TYPE_BUSH) {
+        this->actor.shape.shadowScale = 1.0f;
+        this->actor.shape.shadowAlpha = 60;
+    } else {
+        this->actor.shape.shadowScale = 0.9f;
+        this->actor.shape.shadowAlpha = 70;
+    }
+
+    if (this->actor.shape.rot.y == 0) {
+        this->actor.shape.rot.y = ((u32)Rand_Next() >> 0x10);
+        this->actor.home.rot.y = this->actor.shape.rot.y;
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+    }
+    if (EnKusa_SnapToFloor(this, globalCtx, 0.0f) == 0) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    if (EnKusa_GetWaterBox(this, globalCtx)) {
+        this->isInWater |= 1;
+    }
+
+    this->objIndex = Object_GetIndex(&globalCtx->objectCtx, objectIds[(GET_KUSA_TYPE(&this->actor))]);
+    if (this->objIndex < 0) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    EnKusa_SetupWaitObject(this);
+    if (D_809366B4) {
+        D_80936CD8 = ((u32)Rand_Next() >> 0x10);
+        D_80936CDA = ((u32)Rand_Next() >> 0x10);
+        D_80936CDC = ((u32)Rand_Next() >> 0x10);
+        D_80936CDE = ((u32)Rand_Next() >> 0x10);
+        D_80936CE0 = ((u32)Rand_Next() >> 0x10);
+        D_809366B4 = false;
+        EnKusa_Sway();
+        kusaGameplayFrames = globalCtx->gameplayFrames;
+    }
+    this->kusaMtxIdx = D_809366B0 & 7;
+    D_809366B0++;
+}
+
+void EnKusa_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    EnKusa* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnKusa_SetupWaitObject(EnKusa* this) {
+    this->actionFunc = EnKusa_WaitObject;
+}
+
+/**
+ * @brief Waits for object to be loaded then draws the appropriate model
```

Comment:
This function doesn't actually draw anything, though. I think better to just remove this one too.

Reply:
While it doesn't draw, it calls `this->actor.draw` so would a better name be `EnKusa_SetupDraw`?